File: TemaProiect/motoProject/src/main/java/org/example/motoproject/repos/AppUserRepo.java
package org.example.motoproject.repos;

import java.sql.*;
import java.util.Optional;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class AppUserRepo implements AppUserRepoInterface {
    private final String url;
    private static final Logger logger = LogManager.getLogger(AppUserRepo.class);

    AppUserRepo(String url) {
        this.url = url;
    }

    @Override
    public Optional<String> findPasswordByUsername(String username) {
        String query = "SELECT password FROM users WHERE username = ?";
        try (Connection conn = DriverManager.getConnection(url);
             PreparedStatement stmt = conn.prepareStatement(query)) {
            logger.info("Successfully connected to the database!");
            stmt.setString(1, username);
            ResultSet rs = stmt.executeQuery();
            if (rs.next()) {
                logger.info("Password found!");
                return Optional.of(rs.getString("password"));
            }
        } catch (SQLException e) {
            logger.error("Error connecting: {}", e.getMessage(), e);
            return Optional.empty();
        }
        return Optional.empty();
    }
}

File: TemaProiect/motoProjectC#/motoProjectCSharp/motoProjectCSharp/repos/AppUserRepo.cs
namespace motoProjectCSharp.repos;
using System;
using Microsoft.Data.SqlClient;
using log4net;

public class AppUserRepo : IAppUserRepo
{
    private readonly string _url;
    private static readonly ILog Logger = LogManager.GetLogger(typeof(AppUserRepo));

    public AppUserRepo(string url)
    {
        _url = url;
    }

    public string? FindPasswordByUsername(string username)
    {
        string query = "SELECT password FROM users WHERE username = ?";
        try
        {
            using (var conn = new SqlConnection(_url))
            {
                conn.Open();
                Logger.Info("Successfully connected to the database!");
                using (var stmt = new SqlCommand(query, conn))
                {
                    stmt.Parameters.AddWithValue("@username", username);
                    var result = stmt.ExecuteScalar();
                    if (result != null)
                    {
                        Logger.Info("Password found!");
                        return result.ToString();
                    }
                }
            }
        }
        catch (Exception e)
        {
            Logger.Error($"Error connecting: {e.Message}", e);
            return null;
        }

        return null;
    }
}